bravado/response.py

```diff
@@ -2,16 +2,47 @@
 
 
 class BravadoResponse(object):
-    def __init__(self, incoming_response, result):
-        self.incoming_response = incoming_response
+    """Bravado response object containing the swagger result as well as response metadata.
+
+    WARNING: This interface is considered UNSTABLE. Backwards-incompatible API changes may occur;
+    use at your own risk.
+    """
+
+    def __init__(self, result, response_metadata):
         self.result = result
-        self.response_metadata = BravadoResponseMetadata(
-            incoming_response=self.incoming_response,
-            swagger_result=self.result,
-        )
+        self.response_metadata = response_metadata
+
+    @property
+    def incoming_response(self):
+        return self.response_metadata.incoming_response
 
 
 class BravadoResponseMetadata(object):
-    def __init__(self, incoming_response, swagger_result):
+    """HTTP response metadata.
+
+    NOTE: The `elapsed_time` attribute might be slightly lower than the actual time spent since calling
+    the operation object, as we only start measuring once the call to `HTTPClient.request` returns.
+    Nevertheless, it should be accurate enough for logging and debugging, i.e. determining what went
+    on and how much time was spent waiting for the response.
+
+    WARNING: This interface is considered UNSTABLE. Backwards-incompatible API changes may occur;
+    use at your own risk.
+    """
+
+    def __init__(self, incoming_response, swagger_result, is_fallback_result, elapsed_time, exc_info):
```

Comment:
I would suggest to not have `is_fallback_result` parameter, but rather having the metadata class to evaluate it.
Saying this because otherwise we'll add potentially a lot of new parameters for information that we want to expose with the metadata.
for example we can evaluate `is_fallback_result` as `exc_info != (None, None, None)`

Reply:
I wasn't sure if these two conditions would always be in sync, but you're right, for now they should be. I've changed it.